jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -70,23 +76,43 @@ def _parse_image(app, file_obj, data_label, show_in_viewer, ext=None):
     if isinstance(file_obj, fits.HDUList):
         if 'ASDF' in file_obj:  # JWST ASDF-in-FITS
             if HAS_JWST_ASDF:
-                data_iter = _jwst_to_glue_data(file_obj, ext, data_label)
+
+                # Load all extensions
+                if ext is not None and ext in ['*']:
+                    data_iter = _jwst_all_to_glue_data(file_obj, data_label)
+
+                # Load only specified extension
+                else:
+                    data_iter = _jwst_to_glue_data(file_obj, ext, data_label)
+
+                    # if more than one viewable extension is found in the file,
+                    # issue info message.
+                    _info_nextensions(app, file_obj)
+
             else:
                 raise ImportError('jwst package is missing')
 
-        elif ext is not None:  # Load just the EXT user wants
+        elif ext is not None and ext in ['*']:  # Load all extensions
+            data_iter = _hdus_to_glue_data(file_obj, data_label)
+
+        elif ext is not None and ext not in ['*']:  # Load just the EXT user wants
```

Comment:
By the time this logic is checked, we don't have to worry about `'*'` anymore.
```suggestion
        elif ext is not None:  # Load just the EXT user wants
```